### PR TITLE
test(patterns): measure Topics lift reads, graph size, and timing in a browser

### PR DESCRIPTION
**Pull request by mpsalisbury**
## Why

T0 of the Topics computation-cost plan measures in two tiers, and the browser tier needs the same per-lift numbers the headless probe (#7484) produces. That means body reads and time for the board's cross-reference pivot and for each topic's backlinks, comment count and last activity, plus scheduler graph size and rendering time. No browser helper attributes reads to a named lift. A report keyed by line number would stop matching a candidate as soon as its sources moved, so the helper finds each lift by name. T0 PR 6 extends the scale and navigation benchmarks to call it.

A script that regenerates the core tests' fixture is not part of this PR. As written it was a top-level script whose last act rewrote a tracked file, so no test could import it, and the coverage ratchet counted every line of it as uncovered. It needs a testable shape and a review of its own, and follows separately; `docs/development/BENCHMARKS.md` says how the fixture is recorded meanwhile.

## What Changed

- **`packages/patterns/integration/topics-browser-measurement.ts`**
  - **Compiling the program.** Before measuring, the caller compiles the Topics program the board was seeded from, rooted at `packages/patterns`, the way `cf check` compiles it. It then passes the result to each measurement.
  - **`measureTopicsReads()` samples one operation the caller drives.**
    - It turns telemetry and body read accounting on only around the operation, and waits for a settled view over an idle runtime.
    - It sums `scheduler.run.complete` markers into one row for the producer (`crossrefTable`), one row each for the consumers (`backlinksOf`, `presentCommentCountOf`, `lastActivityOf`), and a remaining total.
    - It also records graph size before and after, elapsed time, and main-thread and worker timing over the same span.
  - **`timeTopicsOperation()` times an interval with accounting off.**
    - It turns telemetry and read accounting off explicitly first, and records that it did.
    - It fails when the worker times no action run, unless the caller declares the operation may run nothing.
    - It can't observe commit errors without telemetry. Its note says a measured sample of the same operation, paired with it, checks them.
  - **Both samplers are bounded to one runtime client.** Each records the runtime client it starts with and throws if that client is replaced mid-operation. Neither brackets cold initialization. Both release their page-side sample entry on every exit.
  - **Errors stay visible.** A measured operation fails when it produces no runs with a read sample, has an event commit error, or raises a page error. If stopping the sample also fails, the operation's own error is kept first.
- **`packages/patterns/integration/topics-browser-measurement-core.ts`** (new, its only import the TypeScript parser): the logic that needs no browser.
  - **Finding a lift.** It parses the authored module and the emitted one with the TypeScript parser, takes the one `const <name> = lift(...)` declaration in each, and matches the position where that call's first argument starts against a running `src`. That argument is the function for the one-argument form the Topics lifts are written as; for `lift(<schema>, <function>)` it is the schema's position, which no run carries, and the comment and the document say so. In the emitted module the callee has to be the compiler's `(0, <alias>.lift)`, left operand included, so `(1, m.lift)` is not one. Nothing decides what is code and what is a string, a comment, a template literal, or a regular expression by hand. `packages/patterns` declares `typescript` at the workspace pin, `npm:typescript@6.0.3`, which ten workspace members now share.
  - **Checking it is the same program.** Running `src` values have the form `cf:module/<identity><path>:<line>:<col>`. Each identity must equal the identity that compiling the Topics sources produces. A mismatch fails and names both identities.
  - **Checking coverage is off.** The page's runtime must report no coverage collector, and no preview may contain a coverage hit call. Otherwise the measurement fails and names coverage.
  - **Checking the code matches exactly.** Each named lift's graph-snapshot preview must equal the first 200 characters of that lift's compiled text. A mismatch fails and shows the first difference.
  - **Not running.** A lift counts as not running only when its module hasn't started and no running `src` names that file under any path.
  - **Unattributable samples.** A sample fails when its runs with reads carry no parseable `src`, or when the board's module isn't running.
  - **All of the above in one place.** `confirmSampledLifts()` runs every check a measured sample's attribution rests on, so a plain `deno test` covers them rather than only the browser test.
- **`packages/patterns/test/topics-browser-measurement-core.test.ts`** (new): unit tests for the core module that run under `deno task test` with no browser or servers. They use recorded compiler output in `topics-browser-measurement-core.fixture.json`, so a later Topics source change doesn't break them; `docs/development/BENCHMARKS.md` says how that recording is taken again. One case pins the recording's extent, so the cases iterating it cannot pass vacuously. Near-miss cases pin what the exact comparison rejects: compiled text one character short, with a trailing space, with one letter's case flipped, with its whitespace collapsed, and an identity that differs from the compiled one in its last character alone.
- **`packages/patterns/integration/topics-browser-measurement.test.ts`**
  - Uses a 2-topic board where one topic cites the other.
  - Opening a topic and returning to the board runs each of the four lifts. Each lift records positive proxy accesses, and each row names the implementation it counted.
  - An operation that runs nothing fails.
  - The timed sampler is also tested: it turns accounting off, fails on zero runs, and accepts a declared no-op.
- **`docs/development/BENCHMARKS.md`:** a section on the helper, covering what it records, what it doesn't, and how the core tests' fixture is recorded again — the one command that prints the compiled half, where a lift's function sits in its output, and, since no command prints a preview, what a preview is and which running board reports it. Timing rows can include subscription and storage durations, though no network bytes are counted. Only `vdom-applicator/apply-batch` isolates rendering.
- **`docs/development/DEPENDENCIES.md` and `packages/patterns/deno.jsonc`:** the TypeScript section covers the tenth member and the second pin. `packages/patterns/test-import-map.json` replaces the package's import map rather than extending it, so lane 1 resolves `typescript` through it; `deno outdated` rewrites the `deno.jsonc` pin alone, and the roll procedure now carries the hand edit of the other. The `deno.jsonc` entry names the second pin, since JSON holds no comment.

## Test plan

The implementer ran these on `259d785593`:
- **Repo checks.** `deno fmt --check` and `deno lint` over the repository, `deno task check` over the workspace, `check-no-waitfor`, `check-docs`, `check-single-copy-deps`, `check-unused-deps`, and `check-deno-pins` all pass.
- **Unit tests.** The core module's 67 test steps pass under the package's lane-1 flags; the package's whole lane-1 suite (66 tests, 313 steps) and its source-coverage and focused-browser passes are green. `git status` is identical before and after.
- **The callee check.** Removing the literal-`0` test from `isCompiledLiftCallee()` turns the new case red: `(1, m.lift)` and `(count(), m.lift)` are read as lift declarations without it.
- **The fixture's extent.** Dropping `lastActivityOf` from the recording leaves both cases that iterate `fixture.lifts` green and turns only the new extent case red.
- **The documented recording procedure.** Each step was run on this checkout: the command prints five `// cf:module/<identity>` headers whose two `/topics/` identities equal what `prepareTopicsProgram()` and a headless board both report; `compiledLiftText()` over that output returns text whose first 200 characters equal the preview a headless board reports, for a truncated lift and a whole one; and adding 68 lines at the top of `topics/main.tsx` changes the identity the same command reports.
- **Near-miss mutations.** Each of these five weakenings of the comparison turns exactly one near-miss test red: a prefix match, a trim before comparing, a case-insensitive compare, a whitespace-normalizing compare, and an eight-character identity check.
- **Deletion mutations.** Deleting `confirmSampledLifts()`'s `requireSameProgram()` call turns the identity near-miss red; deleting its preview coverage check turns the coverage case red, which then reports a text mismatch instead of coverage.
- **Browser test.** 12 steps pass against local servers, including the case with pattern coverage on, which fails with the coverage message.

The browser integration lane was not re-run for the last two rounds. Neither touched a code path it exercises: one added the literal-`0` test to the core module's callee predicate, which lane 1 covers against the recorded compiled text, and the other deleted a standalone tool and the unused `rewrite` option of the helper's private compile.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_014hGt1ENr9iDzhLhrgEy1Ep

### CODE REVIEW

File: packages/patterns/integration/topics-browser-measurement.ts [hunk 1]
```diff
@@ -0,0 +1,833 @@
...
+      totals.distinctDocuments += marker.reads.distinctDocuments;
+      totals.registeredDependencies += marker.reads.registeredDependencies;
+    };
+    rt.on("telemetry", listener);
+    (scope.__cfTopicsSamplings ??= {})[token] = {
+      sampling,
+      stop: () => rt.off("telemetry", listener),
+    };
+    await rt.setTelemetryEnabled(true);
+    await rt.setReadStatsEnabled(true);
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply packages/patterns/integration/topics-browser-measurement.ts]: Fixed in add636d097. If turning on telemetry or read accounting fails, the page-side setup now unsubscribes the listener, deletes the sampling entry, tries to turn both flags off, and rethrows the original error (topics-browser-measurement.ts, the catch in the start-sampling evaluate).

File: packages/patterns/integration/topics-browser-measurement.test.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      const sample = await measureTopicsReads(session.page, {
+        label: "open topic and return",
+        operation: openTopicAndReturn,
+      });
+      console.error(formatTopicsSample(sample).join("\n"));
+
+      expect(sample.lifts.map((lift) => lift.name)).toEqual(
+        TOPICS_LIFTS.map((lift) => lift.name),
+      );
+      expect(
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply packages/patterns/integration/topics-browser-measurement.test.ts]: Fixed in add636d097. The browser test now fails any named lift that has zero runs or zero proxy accesses (topics-browser-measurement.test.ts:133), so broken read accounting cannot pass. Each row also carries the implementation preview it counted, which the test checks against its own lift.

File: docs/development/BENCHMARKS.md [hunk 3]
```diff
@@ -313,6 +313,61 @@ run stays green.
...
+
+- Transaction-attempt reads. The runtime client's read-stats request enables
+  body accounting only, so attempt reads come from the headless tier.
+- Rendering beyond VDOM application. The main thread's
+  `vdom-applicator/apply-batch` timing, which covers applying worker VDOM
+  batches to the DOM, is the only rendering time the shell records. Lit element
+  updates, style, layout, and paint have no timing of their own and appear only
+  in the elapsed time. The main thread's `runtime-client/ipc/*` rows are waits
+  on the worker, not main-thread work.
+- Network bytes, subscriptions, and storage work outside reactive bodies.
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply docs/development/BENCHMARKS.md]: Fixed in add636d097. BENCHMARKS.md now says the helper does not count network bytes or subscription events and does not attribute storage work as body reads, while its timing rows can include the durations of subscription requests and of worker storage spans.

File: docs/development/BENCHMARKS.md [hunk 4]
```diff
@@ -313,6 +313,61 @@ run stays green.
 Because that ceiling is the board's cost and not a property of the benchmark,
 the two skipped sizes should be enabled as part of whatever lowers it.
 
+## Topics browser measurement
+
+`packages/patterns/integration/topics-browser-measurement.ts` measures one
+operation a caller drives in a Topics board's browser page, for the browser tier
+of [the Topics computation plan](../plans/topics-computation-cost.md). It adds
+no benchmark series of its own; a board benchmark calls it around a workload.
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply docs/development/BENCHMARKS.md]: Fixed in add636d097. The section now describes a helper a caller invokes around an operation. It adds no benchmark series of its own, and the plan's T0 work wires it into the scale and navigation benchmarks.

File: packages/patterns/integration/topics-browser-measurement.ts [hunk 5]
```diff
@@ -0,0 +1,833 @@
...
+ */
+export async function timeTopicsOperation(
+  page: Page,
+  options: TimeTopicsOperationOptions,
+): Promise<TopicsTimedSample> {
+  const pageErrors = watchPageErrors(page);
+  try {
+    const before = await readRuntimeState(page);
+    const startedAt = performance.now();
+    options.interval?.start();
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply packages/patterns/integration/topics-browser-measurement.ts]: Fixed in add636d097. timeTopicsOperation() now starts the interval, then runs the operation and settles inside a try whose finally calls options.interval?.end(), so a rejected operation or settle still ends the interval.

File: packages/patterns/integration/topics-browser-measurement-core.ts [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+): boolean {
+  const unaliased = preview
+    .replace(/\(0,\s*[A-Za-z_$][\w$]*\.([A-Za-z_$][\w$]*)\)/g, "$1")
+    .replace(/\b[A-Za-z_$][\w$]*_\d+\.(?=[A-Za-z_$])/g, "");
+  const wanted = identifiersOf(unaliased);
+  if (preview.length >= PREVIEW_LENGTH) wanted.pop();
+  const available = identifiersOf(declaration);
+  let next = 0;
+  for (const identifier of wanted) {
+    while (next < available.length && available[next] !== identifier) next++;
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply packages/patterns/integration/topics-browser-measurement-core.ts]: Fixed in 0055c6295a. implementationMatches() now compares the preview and the declaration token by token (identifiers, operators and punctuation, literals). The only differences it allows are declaration type syntax, the compiler's import alias, trailing commas, and the preview's cut. A complete preview must end where the declared function ends, so an identifier-prefix implementation no longer matches. The core unit tests cover this, including a function whose tokens are a prefix of the declared one.

File: packages/patterns/integration/topics-browser-measurement.ts [hunk 7]
```diff
@@ -0,0 +1,854 @@
...
+ * `token`, disables read accounting and telemetry, and returns what it
+ * collected.
+ *
+ * @throws If the runtime client is not the one the sampling started against:
+ *   its listener and its accounting belonged to the replaced client.
+ */
+async function stopSampling(page: Page, token: string): Promise<PageSampling> {
+  return await page.evaluate(async (token: string) => {
+    const scope = globalThis as MeasurementGlobal;
+    const entry = scope.__cfTopicsSamples?.[token];
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply packages/patterns/integration/topics-browser-measurement.ts]: Fixed in 0055c6295a. Both measureTopicsReads() and timeTopicsOperation() now release their page-side sample entry on every exit path through releaseSample(), keeping the operation's own error first. Browser cases for a failing operation, a no-run failure, and a replaced runtime client each assert that no sample is left in the page.

File: packages/patterns/integration/topics-browser-measurement-core.ts [hunk 8]
```diff
@@ -0,0 +1,735 @@
...
+export const MIN_CUT_PREVIEW_BODY_TOKENS = 8;
+
+/** Matches a run's `src`: the module identity, then `/<path>:<line>:<col>`. */
+const SRC_PATTERN = /^cf:module\/([^/]+)(\/.+:\d+:\d+)$/;
+
+/**
+ * Matches a line that starts at column 0 with anything but a closing bracket,
+ * which in module-scope code starts a statement or a comment of its own.
+ */
+const TOP_LEVEL_START = /(?<=\n)[^\s)\]}]/;
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply packages/patterns/integration/topics-browser-measurement-core.ts]: No longer applicable as of 97b3cb7955. The helper no longer finds where a lift's declaration ends: locateLift() returns only the lift's start position, and a lift is confirmed by module identity plus an exact comparison of its compiled text, so no column-0 boundary rule remains.

File: packages/patterns/integration/topics-browser-measurement-core.ts [hunk 9]
```diff
@@ -0,0 +1,744 @@
...
+      const close = text.indexOf("*/", at + 2);
+      at = close === -1 ? text.length : close + 2;
+      continue;
+    }
+    let end: number;
+    if (char === '"' || char === "'") {
+      end = quotedEnd(text, at);
+    } else if (char === "`") {
+      end = templateEnd(text, at);
+    } else if (char === "/" && startsRegex(previous)) {
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply packages/patterns/integration/topics-browser-measurement-core.ts]: Fixed in 7258918043. `startsRegex()` and the whole hand-written token scanner are gone: `compiledLiftText()` now parses the emitted module with the TypeScript parser (`packages/patterns/integration/topics-browser-measurement-core.ts`, `parse()` and `liftArguments()` under "Parsing"), finds the one `const <name> = (0, <alias>.lift)(<function>, ...)` declaration in the tree, and slices the text between the first argument's start and end. The parser decides what a `/` is, so a postfix `++`/`--` before a division is no longer read as the start of a regular expression.

A unit test pins it: "returns a function whose body divides twice after a postfix increment" in `packages/patterns/test/topics-browser-measurement-core.test.ts`. Its body is `counts[index++] / (counts.length / 2)`, which the old scanner swallowed as a regular expression literal — taking the `(` with it, so the bracket depth never came back to zero and the lift's argument was reported as never closing.

File: packages/patterns/integration/topics-browser-measurement-core.ts [hunk 10]
```diff
@@ -0,0 +1,744 @@
...
+  js: string,
+  name: string,
+  source = "the compiled module",
+): string {
+  requirePlainName(name);
+  const declaration = new RegExp(
+    String.raw`\bconst\s+${name}\s*=\s*\(0,\s*[A-Za-z_$][\w$]*\.lift\)\(`,
+    "g",
+  );
+  const matches = [...js.matchAll(declaration)];
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply packages/patterns/integration/topics-browser-measurement-core.ts]: Fixed in 7258918043. The raw declaration regex is gone from both lookups. `compiledLiftText()` and `locateLift()` parse the module — `ts.ScriptKind.JS` for the emitted one, `ts.ScriptKind.TSX` for the authored one — and count `VariableDeclaration` nodes whose name is the lift's and whose initializer is the lift call (`liftArguments()` and `isCompiledLiftCallee()` under "Parsing" in `packages/patterns/integration/topics-browser-measurement-core.ts`). Declaration-shaped text inside a string, a comment, a template literal, or a regular expression is part of that literal or comment in the tree, so it is not a second declaration.

Two unit tests pin it in `packages/patterns/test/topics-browser-measurement-core.test.ts`: "reads no second declaration out of a string or a comment in the function" for the compiled side, where the old regex counted three declarations and aborted, and "reads no declaration out of a string, a comment, a template literal, or a regular expression" for the authored side.

File: docs/development/BENCHMARKS.md [hunk 11]
```diff
@@ -313,6 +313,112 @@ run stays green.
...
+whose worker collects pattern coverage fails it with a message naming coverage;
+a runtime client replaced during a measured operation fails it after telemetry
+and read accounting are turned off on the client they were turned on in; and the
+timed sampler turns off telemetry a caller left on. The decisions that need no
+browser live in `topics-browser-measurement-core.ts` beside the helper, and
+`packages/patterns/test/topics-browser-measurement-core.test.ts` tests them
+under a plain `deno test`, partly against compiled text, previews, and module
+identities recorded from the Topics sources in a fixture beside that test.
+`packages/patterns/tools/regenerate-topics-measurement-fixture.ts` rewrites
+everything in that fixture a compile produces, and fails when a preview
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply docs/development/BENCHMARKS.md]: Valid, and the claim is now qualified (ab790dc16d).

`docs/development/BENCHMARKS.md` keeps the sentence about a preview recorded from a browser and adds the exception: the one field the generator carries over without a check is the preview recorded from a compile with pattern coverage on, because the tests read it only for the coverage hit call it holds, which a compile with coverage off never writes, and a recorded hit call stays one however the Topics sources move.

The same qualification is in the two other places that carried the claim: the module comment of `packages/patterns/tools/regenerate-topics-measurement-fixture.ts` (and the doc comment on its `instrumentedPreview` field), and the header of `packages/patterns/test/topics-browser-measurement-core.test.ts`. The cases that read the field are `requireUninstrumentedPreviews() > throws naming pattern coverage for a preview holding a coverage hit call` and `confirmLiftImplementations() > throws naming coverage, not the text, for an implementation carrying coverage instrumentation`; both assert on the hit call, neither on the field being current.

File: packages/patterns/integration/topics-browser-measurement-core.ts [hunk 12]
```diff
@@ -0,0 +1,621 @@
...
+ */
+function isCompiledLiftCallee(callee: ts.Expression): boolean {
+  if (!ts.isParenthesizedExpression(callee)) return false;
+  const comma = callee.expression;
+  return ts.isBinaryExpression(comma) &&
+    comma.operatorToken.kind === ts.SyntaxKind.CommaToken &&
+    ts.isPropertyAccessExpression(comma.right) &&
+    ts.isIdentifier(comma.right.expression) &&
+    comma.right.name.text === "lift";
+}
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply packages/patterns/integration/topics-browser-measurement-core.ts:628]: Valid, and taken as suggested (ab790dc16d). `isCompiledLiftCallee()` now requires the left operand to be the literal `0`:

```ts
    ts.isNumericLiteral(comma.left) && comma.left.text === "0" &&
```

Its doc comment states the requirement, so the comment and the check agree.

Pinned by a new case in `packages/patterns/test/topics-browser-measurement-core.test.ts`, `compiledLiftText() > reads no declaration out of a comma expression whose left operand is not `0``, which feeds `compiledLiftText()` both `(1, commonfabric_2.lift)((value) => value)` and `(count(), commonfabric_2.lift)((value) => value)` and expects `found 0` for each. Deleting the line above turns that case red, and no other case in the file.

File: packages/patterns/integration/topics-browser-measurement-core.ts [hunk 13]
```diff
@@ -0,0 +1,628 @@
...
+  isLift: (callee: ts.Expression) => boolean,
+): ts.Expression[] {
+  const found: ts.Expression[] = [];
+  const visit = (node: ts.Node): void => {
+    if (
+      ts.isVariableDeclaration(node) && ts.isIdentifier(node.name) &&
+      node.name.text === name && node.initializer !== undefined &&
+      ts.isCallExpression(node.initializer) &&
+      isLift(node.initializer.expression) &&
+      node.initializer.arguments.length > 0
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply packages/patterns/integration/topics-browser-measurement-core.ts:602]: Valid, and confirmed against the code: `liftArguments()` tests `ts.isVariableDeclaration(node)` with no `ts.NodeFlags.Const` check, while five places say `const` — `locateLift`'s doc comment (:87), its thrown message (:113), `compiledLiftText`'s doc (:126), `liftArguments`' own doc (:586), and `docs/development/BENCHMARKS.md:353`.

It cannot misattribute a measurement, which is why it is not being fixed in this pull request. A lone `let`-declared lift is found and measured correctly. A `let` shadowing a `const` of the same name gives `found.length === 2`, and `locateLift` throws rather than picking one. So the divergence is between the comments and the predicate, not in what the helper reports.

It is the fifth finding of one shape on this branch: a sentence that spells the accepted form more narrowly than the code accepts it. The other four were corrected in the round that produced this head. Rather than correct a fifth instance and invite a sixth, the general form goes into a follow-up pull request of its own: make every sentence in this file that names the accepted declaration form say the same true thing, and decide once whether the predicate should require `const` — which would make the existing sentences true instead, and is probably the better answer, since these helpers only ever measure hoisted `const` lifts.

Carried there with two siblings this review found: the thrown message above, and `compiledLiftText`'s doc, which says it "returns the function the compiled module passes to the lift" when it returns argument zero.

File: docs/development/BENCHMARKS.md [hunk 14]
```diff
@@ -313,6 +313,131 @@ run stays green.
...
+it, which is what lets those cases hold still as the sources move. It is taken
+again by hand, when a case needs material the recording does not hold. The
+compiled half comes from one command:
+
+```bash
+deno task cf check packages/patterns/topics/main.tsx --json --no-check \
+  --root packages/patterns
+```
+
+It emits the whole compiled program, each module preceded by the
```
[the remark this answers is not in the document]
Comment (mpsalisbury) [reply docs/development/BENCHMARKS.md]: Not valid, and I checked it against a run rather than against the help text.

Running the documented command on this checkout, the output's `files[0].output` string carries five `// cf:module/<identity>` header lines, at lines 1, 1712, 1804, 1974 and 8247:

```
   1 // cf:module/HUSIEdRvkffqvPjUUXMmNfTqiXVR-zvlVmDrbzd_xZw
1712 // cf:module/wTjSCz1KauvyrB3vk0Z4vK5FQJHgfr6o-aBdSn7dXTg
1804 // cf:module/Clavv6LzsaJil48dQZ-rrOnnix2CQchFZ-3W_47N_rI
1974 // cf:module/3GFMuKDUhQbC3vKwKrzhkOthZqiQh4T-wydpNByacF0
8247 // cf:module/uvM3GfBaAVyR2xHI0wPhiGXPSgoo8SqQP23RMRjX5eQ
```

Those are content identities, not specifiers, and they are the same ones the runtime reports. Two independent routes agree with the two `/topics/` identities above:

- `prepareTopicsProgram()`'s own compile reports `/topics/main.tsx` = `HUSIEd…` (line 1) and `/topics/topic.tsx` = `3GFMuK…` (line 1974).
- A headless board built from `topics-headless-fixture.ts` reports its lifts' `src` as `cf:module/HUSIEdRvkffqvPjUUXMmNfTqiXVR-zvlVmDrbzd_xZw/topics/main.tsx:341:2` and `cf:module/3GFMuKDUhQbC3vKwKrzhkOthZqiQh4T-wydpNByacF0/topics/topic.tsx:1714:25`, which is the `<identity>` `parseSrc()` reads and `confirmSampledLifts()` compares against.

So a recorder can take both recorded identities from this one command, which is what the paragraph relies on.

Your line number did point at real problems in the same paragraph, and 259d785593 fixes them: it said a lift's compiled text was its `const <name> = (0, <alias>.lift)(...)` declaration when `compiledLiftText()` returns the function that declaration is passed; it named a provenance for the previews but no mechanism; and it referred to a line count without saying the lines go into the source before the command runs. The paragraph now also says the command prints a JSON document whose `files[0].output` holds the program, since a hand-recorder needs to know to extract it.
